src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -2096,6 +2114,142 @@ def set_attention_slice(self, slice_size: Optional[int]):
         for module in modules:
             module.set_attention_slice(slice_size)
 
+    @classmethod
+    def from_pipe(cls, pipeline: "DiffusionPipeline", **kwargs) -> "DiffusionPipeline":
+        r"""
+        Create a new pipeline from a given pipeline. This method is useful to create a new pipeline with the same
+        weights and configurations without reallocating additional memory.
+
+        Arguments:
+            pipeline (`DiffusionPipeline`):
+                The pipeline from which to create a new pipeline.
+
+        Returns:
+            `DiffusionPipeline`:
+                A new pipeline with the same weights and configurations as `pipeline`.
+
+        Examples:
+
+        ```py
+        >>> from diffusers import StableDiffusionPipeline
+
+        >>> pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
+        >>> new_pipe = StableDiffusionPipeline.from_pipe(pipe)
+        ```
+        """
+
+        if hasattr(pipeline, "_all_hooks") and len(pipeline._all_hooks) > 0:
```

Comment:
this is the part we make sure if the pipeline has previous called `enable_model_cpu_offload`, it will still work properly with `from_pipe`